lib/Sema/CSDiag.cpp

```diff
@@ -845,6 +845,15 @@ namespace {
     evaluateCloseness(DeclContext *dc, Type candArgListType, ArrayRef<CallArgParam> actualArgs);
       
     void filterList(ArrayRef<CallArgParam> actualArgs);
+#if _MSC_VER
+    // MSVC confused when choosing a correct method.
+    void filterList(SmallVector<CallArgParam, 4> actualArgs) {
+      return filterList(ArrayRef<CallArgParam>(actualArgs));
+    }
+    void filterList(CallArgParam actualArgs) {
+      return filterList(ArrayRef<CallArgParam>(actualArgs));
+    }
+#endif
```

Comment:
Please don't add #ifdefs for this.  If there is a problem resolving the overload, just rename the methods to be distinct, and update the callers.